clients/stellar-relay-lib/src/connection/flow_controller.rs

```diff
@@ -1,33 +1,83 @@
-use substrate_stellar_sdk::types::MessageType;
+use substrate_stellar_sdk::types::{MessageType, StellarMessage, SendMoreExtended, SendMore};
+use crate::connection::handshake::AUTH_FLAG;
+
+
+pub const MAX_FLOOD_MSG_CAP: u32 = 200;
+pub const FLOW_CONTROL_SEND_MORE_BATCH_SIZE: u32 = 40;
+pub const PER_FLOOD_READING_CAPACITY_BYTES: u32 = 300000;
+pub const FLOW_CONTROL_SEND_MORE_BATCH_SIZE_BYTES: u32 = 100000;
+
 
-pub const MAX_FLOOD_MSG_CAP: u32 = 2000;
 
 #[derive(Debug, Default)]
 pub struct FlowController {
-	enabled: bool,
-	flood_msg_cap: u32,
+	pub(crate) flow_control_bytes_enabled: bool,
+	messages_received_in_current_batch: u32,
+	bytes_received_in_current_batch: u32,
 }
 
 impl FlowController {
-	pub fn enable(&mut self, local_overlay_version: u32, remote_overlay_version: u32) {
-		self.enabled = remote_overlay_version >= 20 && local_overlay_version >= 20;
+	pub fn enable_bytes(&mut self, local_overlay_version: u32, remote_overlay_version: u32) {
+		self.flow_control_bytes_enabled = remote_overlay_version >= 28 && local_overlay_version >= 28 && AUTH_FLAG == 200;
```

Comment:
The last check `AUTH_FLAG == 200` is useless if we always define this as a constant. I think we should pass it into this function instead or remove the condition completely

Reply:
True, the reason I left it is because we may want to change the flag back to `100`, which we can do by changing the constant, and will not enable this bytes control.
We can do the fully parametrized implementation like [here](https://github.com/stellarbeat/js-stellar-node-connector/blob/9120f24b75867700ba0ece76369166959205c6b9/src/connection/connection.ts#L634) and it will make sense to compare it against the flag.

Reply:
I removed the check, since we enable anyway flag `200` when local overlay is >= 28, therefore it is redundant.